pkg/controller/job/job_controller.go

```diff
@@ -440,14 +440,14 @@ func (jm *JobController) syncJob(key string) (bool, error) {
 		klog.V(4).Infof("Finished syncing job %q (%v)", key, time.Since(startTime))
 	}()
 
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	tenant, namespace, name, err := cache.SplitMetaTenantNamespaceKey(key)
 	if err != nil {
 		return false, err
 	}
-	if len(ns) == 0 || len(name) == 0 {
+	if len(tenant) == 0 || len(namespace) == 0 || len(name) == 0 {
```

Comment:
I remember tenant could be empty if it's the resource is in system space.  Not sure if it's still the case. If yes, then the check len(tenant)==0 will be a problem.   (ideally obj.tenant should be equal to "system" instead of empty.)

Reply:
Job is a namespace-scoped object. For a namespace-scoped object, its tenant and namespace values in the ObjectMeta cannot be empty.